crates/compiler/gen_dev/src/generic64/mod.rs

```diff
@@ -746,6 +770,40 @@ impl<
                     .load_to_general_reg(&mut self.buf, src2);
                 ASM::imul_reg64_reg64_reg64(&mut self.buf, dst_reg, src1_reg, src2_reg);
             }
+            Layout::Builtin(Builtin::Int(
+                IntWidth::U64 | IntWidth::U32 | IntWidth::U16 | IntWidth::U8,
+            )) => {
+                // TODO find a general way to do this
+                let rax = CC::GENERAL_RETURN_REGS[0];
+                let rdx = CC::GENERAL_RETURN_REGS[1];
+
+                self.storage_manager
+                    .ensure_reg_free(&mut self.buf, RegStorage::General(rax));
+
+                self.storage_manager
+                    .ensure_reg_free(&mut self.buf, RegStorage::General(rdx));
+
+                self.storage_manager
+                    .load_to_specified_general_reg(&mut self.buf, src1, rax);
+
+                let dst_reg = self.storage_manager.claim_general_reg(&mut self.buf, dst);
+                let src2_reg = self
+                    .storage_manager
+                    .load_to_general_reg(&mut self.buf, src2);
+                ASM::umul_reg64_reg64_reg64(&mut self.buf, dst_reg, rax, src2_reg);
```

Comment:
should rdx be the `dst_reg` here?

Reply:
rdx is the overflow register, so it shouldn't be needed at all. Just needs to be cleared to not overwrite a variable. The result will be put in rdx